shared/go/aws/cloudwatchlogs/cloudwatch.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudwatchlogs
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins/source"
+	"github.com/falcosecurity/plugins/shared/go/aws/session"
+)
+
+// Client represents a client for CloudwatchLogs API
+type Client struct {
+	*cloudwatchlogs.CloudWatchLogs
+}
+
+// CreateClient returns a Client for CloudwatchLogs API
+func CreateClient(cfgs *aws.Config) *Client {
+	return &Client{
+		CloudWatchLogs: cloudwatchlogs.New(session.CreateSession(), cfgs),
+	}
+}
+
+// CreateFilter returns a filter for the query to CloudwatchLogs API
+func CreateFilter(startTime, endTime int64, filterPattern, logGroupName, logStreamNamePrefix, logStreamNames string) *cloudwatchlogs.FilterLogEventsInput {
+	return &cloudwatchlogs.FilterLogEventsInput{
+		StartTime:           time.Now().Add(-10 * time.Second).UnixMilli(),
+		FilterPattern:       aws.String(filterPattern),
+		LogGroupName:        aws.String(logGroupName),
+		LogStreamNamePrefix: aws.String(logStreamNamePrefix),
+		LogStreamNames:      aws.String(logStreamNames),
+	}
+}
+
+// Open returns an instance with the functionn called to retrieve logs
+func (client *Client) Open(filters *cloudwatchlogs.FilterLogEventsInput, refresh int64) (source.Instance, error) {
+	eventC := make(chan source.PushEvent)
+
+	go func() {
+		defer close(eventC)
+		for {
+			filters.SetNextToken("")
+			client.getLogEvents(filters, eventC)
+			time.Sleep(time.Duration(refresh) * time.Second)
+		}
+	}()
+	return source.NewPushInstance(eventC)
+}
+
+func (client *Client) getLogEvents(filters *cloudwatchlogs.FilterLogEventsInput, eventC chan source.PushEvent) error {
+	logs, err := client.CloudWatchLogs.FilterLogEvents(filters)
+	if err != nil {
+		return sdk.ErrEOF
+	}
+	if len(logs.Events) == 0 {
```

Comment:
Note that EOF will make the event source terminate, and eventually will make consumers like Falco exit. What if we want the poll for new log entries and wait for new ones to be written? I don't think this should be the termination condition (at least, this should be something configurable from the outside). In general, I think the right error to return here is `sdk.ErrTimeout`.